### PR TITLE
Bump Spring Boot version

### PR DESCRIPTION
**Pull request by sorin-stefan-iovita-sap**
## Description

This PR upgrades to the newest version of the Spring Boot. It also removes dependencies version that already come with this new version of the Spring Boot.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -203,17 +195,6 @@
...
-        <groupId>junit</groupId>
-        <artifactId>junit</artifactId>
-        <version>${junit.version}</version>
-        <scope>test</scope>
-      </dependency>
-      <dependency>
-        <groupId>org.apache.tomcat.embed</groupId>
-        <artifactId>tomcat-embed-core</artifactId>
-        <version>${tomcat-embed-core.version}</version>
-      </dependency>
```
Comment (hilmarf) [inline pom.xml:216]: hmmm... shall we really remove that overriding? I have concerns, that the next security finding will occur in the same libraries... especially tomcat has very frequent new issues...

File: pom.xml [hunk 2]
```diff
@@ -173,11 +170,6 @@
         <version>${guava.version}</version>
         <scope>runtime</scope>
       </dependency>
-      <dependency>
-        <groupId>org.apache.httpcomponents</groupId>
-        <artifactId>httpclient</artifactId>
-        <version>${httpclient.version}</version>
-      </dependency>
```
Comment (hilmarf) [inline pom.xml:180]: as long as it's the same version as spring uses it, we can keep it...

File: pom.xml (hunk 1 above)
Comment (sorin-stefan-iovita-sap) [reply pom.xml:216]: Good point. I'll leave it as it was.

File: pom.xml (hunk 2 above)
Comment (sorin-stefan-iovita-sap) [reply pom.xml]: I just reverted it. 😄

**Review by EugenM-SAP — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

**Pull request merged**